packages/json-schema/test/bundling.test.ts

```diff
@@ -13,86 +13,137 @@ describe("bundling", () => {
+    assert.strictEqual(types.$defs.Bar.$id, "Bar.json");
   });
 
-  it("bundles non-schema dependencies by default", async () => {
+  it("doesn't create bundled schemas for referenced non-JSON Schema types", async () => {
     const schemas = await emitSchema(
       `
       @jsonSchema
-      model test { b: Bar }
-      model Bar { b: Baz }
+      model testModel { a: nonModel, b: nonArray, c: nonUnion, d: nonEnum, e: nonScalar  }
+      @jsonSchema
+      model testArray is Array<nonModel>;
+      @jsonSchema
+      union testUnion { nonModel }
+      @jsonSchema
+      enum testEnum { a, b }
+      @jsonSchema
+      scalar testScalar extends nonScalar;
+      
+      model nonModel { b: Baz }
       model Baz { }
+      model nonArray is Array<nonModel>;
+      scalar nonScalar extends string;
+      enum nonEnum { a, b }
+      union nonUnion { nonModel }
+
     `,
-      {},
-      { emitNamespace: false }
+      { bundleId: "test.json" },
+      {
+        emitNamespace: false,
+        emitTypes: ["testModel", "testArray", "testUnion", "testEnum", "testScalar"],
+      }
     );
 
-    assert.strictEqual(schemas["test.json"].$id, "test.json");
-    assert.strictEqual(schemas["test.json"].$defs.Bar.$id, "Bar.json");
-    assert.strictEqual(schemas["test.json"].$defs.Baz.$id, "Baz.json");
+    const defs = Object.keys(schemas["test.json"].$defs);
+    strictEqual(defs.length, 5);
+    ["testModel", "testArray", "testUnion", "testEnum", "testScalar"].forEach((name) => {
+      assert(defs.indexOf(name) != -1, "defs should contain " + name);
+    });
   });
-
-  it("doesn't bundle non-schema dependencies when passing emitAllRefs", async () => {
+  it("with emitAllRefs, creates bundled schemas for referenced non-JSON Schema types", async () => {
     const schemas = await emitSchema(
       `
       @jsonSchema
-      model test { b: Bar }
-      model Bar { }
+      model testModel { a: nonModel, b: nonArray, c: nonUnion, d: nonEnum, e: nonScalar  }
+      @jsonSchema
+      model testArray is Array<nonModel>;
+      @jsonSchema
+      union testUnion { nonModel }
+      @jsonSchema
+      enum testEnum { a, b }
+      @jsonSchema
+      scalar testScalar extends nonScalar;
+      
+      model nonModel { b: Baz }
+      model Baz { }
+      model nonArray is Array<nonModel>;
+      scalar nonScalar extends string;
+      enum nonEnum { a, b }
+      union nonUnion { nonModel }
+
     `,
-      { emitAllRefs: true },
-      { emitNamespace: false }
+      { bundleId: "test.json", emitAllRefs: true },
+      {
+        emitNamespace: false,
+        emitTypes: ["testModel", "testArray", "testUnion", "testEnum", "testScalar"],
+      }
     );
 
-    assert.strictEqual(schemas["test.json"].$id, "test.json");
-    assert.strictEqual(schemas["test.json"].$defs, undefined);
-    assert.strictEqual(schemas["Bar.json"].$id, "Bar.json");
+    const defs = Object.keys(schemas["test.json"].$defs);
+    strictEqual(defs.length, 11);
```

Comment:
I feel there should be a type that isn't referenced from any `@jsonSchema`-adroned types in this test to validate that such type won't get into the generated JSON Schema at all. Then check that it does get into the scheme with `emitAllModels`

Reply:
Good suggestion, I've added this test.